### PR TITLE
Automate next version bump in cut release

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
calculate next version instead of using workflow input in cut-release-branch.yml

### CODE REVIEW

File: parse-semver/action.yml [hunk 1]
```diff
@@ -7,6 +7,8 @@ inputs:
 outputs:
   version:
     description: "Parsed version"
+  next-minor-version:
```
Comment (emmyoop) [inline parse-semver/action.yml:10]: Should we call this `next-minor-alpha-version` for clarity?

File: .github/workflows/cut-release-branch.yml [hunk 2]
```diff
@@ -294,7 +299,8 @@ jobs:
         run: |
           source env/bin/activate
           python -m pip install -r dev-requirements.txt
-          env/bin/bumpversion --allow-dirty --new-version ${{ inputs.version_to_bump_main }} major
+          env/bin/bumpversion --allow-dirty \ 
+           --new-version : ${{ fromJSON(needs.create_temp_branch.outputs.next-version) }} major
```
Comment (emmyoop) [inline .github/workflows/cut-release-branch.yml:303]: Does the colon do something here?  Also why is next-version not just a string?  Why is it json?

**Review by emmyoop — CHANGES_REQUESTED** [review]
A few questions on implementation.

File: .github/workflows/cut-release-branch.yml [hunk 3]
```diff
@@ -34,7 +34,8 @@ on:
   workflow_call:
     inputs:
       version_to_bump_main:
-        required: true
+        description: "DEPRECATED: The version to bump main to.  This is no longer used."
```
Comment (emmyoop) [inline .github/workflows/cut-release-branch.yml:37]: You should create a new ticket to clean this up and remove it from all the calling workflows in the repos.

File: .github/workflows/cut-release-branch.yml (hunk 2 above)
Comment (colin-rogers-dbt) [reply .github/workflows/cut-release-branch.yml:303]: nope, just forgot to remove after I copied it from somewhere that needed it

File: .github/workflows/cut-release-branch.yml (hunk 3 above)
Comment (colin-rogers-dbt) [reply .github/workflows/cut-release-branch.yml:37]: https://github.com/dbt-labs/actions/issues/115

**Review by emmyoop — APPROVED** [review]
LGTM

**Pull request merged**